tests/Integration/Database/DatabaseEloquentModelAttributeCastingTest.php

```diff
@@ -49,8 +49,8 @@ public function testBasicCustomCasting()
 
         $model = new TestEloquentModelWithAttributeCast;
 
-        $model->address = $address = new AttributeCastAddress('110 Kingsbrook St.', 'My Childhood House');
-        $address->lineOne = '117 Spencer St.';
+        $model->address = new AttributeCastAddress('110 Kingsbrook St.', 'My Childhood House');
+        $model->address->lineOne = '117 Spencer St.';
```

Comment:
Why did you change this test?

Reply:
Apologies, I was sure I had added an edit to the PR description highlighting that by introducing this performance check to avoid multiple calls you then can't modify the object by itself but you need to access the object through the model.

So given
```
$address = new AttributeCastAddress('110 Kingsbrook St.', 'My Childhood House');
$model->address = $address
```
you should do
```
$model->address->lineOne = '117 Spencer St.';
```
not
```
$address->lineOne = '117 Spencer St.';
```

I guess that this introduces a breaking change and then this PR should target the next major version.